crates/xilem_core/src/sequence.rs

```diff
@@ -295,26 +295,36 @@ macro_rules! generate_viewsequence_trait {
-        $crate::impl_view_tuple!($viewseq, $pod, $cx, $changeflags,
-            V0, V1, V2, V3; 0, 1, 2, 3);
-        $crate::impl_view_tuple!($viewseq, $pod, $cx, $changeflags,
-            V0, V1, V2, V3, V4; 0, 1, 2, 3, 4);
-        $crate::impl_view_tuple!($viewseq, $pod, $cx, $changeflags,
-            V0, V1, V2, V3, V4, V5; 0, 1, 2, 3, 4, 5);
-        $crate::impl_view_tuple!($viewseq, $pod, $cx, $changeflags,
-            V0, V1, V2, V3, V4, V5, V6; 0, 1, 2, 3, 4, 5, 6);
-        $crate::impl_view_tuple!($viewseq, $pod, $cx, $changeflags,
-            V0, V1, V2, V3, V4, V5, V6, V7; 0, 1, 2, 3, 4, 5, 6, 7);
-        $crate::impl_view_tuple!($viewseq, $pod, $cx, $changeflags,
-            V0, V1, V2, V3, V4, V5, V6, V7, V8; 0, 1, 2, 3, 4, 5, 6, 7, 8);
-        $crate::impl_view_tuple!($viewseq, $pod, $cx, $changeflags,
-            V0, V1, V2, V3, V4, V5, V6, V7, V8, V9; 0, 1, 2, 3, 4, 5, 6, 7, 8, 9);
+        $crate::impl_view_tuple!($viewseq <$($tp,)*>, $pod, $cx, $changeflags, ;);
+        $crate::impl_view_tuple!($viewseq <$($tp,)*>, $pod, $cx, $changeflags,
+            V0<$($tp,)*>;
+            0);
+        $crate::impl_view_tuple!($viewseq <$($tp,)*>, $pod, $cx, $changeflags,
+            V0<$($tp,)*>, V1<$( $tp, )*>;
+            0, 1);
+        $crate::impl_view_tuple!($viewseq <$($tp,)*>, $pod, $cx, $changeflags,
+            V0<$($tp,)*>, V1<$($tp,)*>, V2<$($tp,)*>;
+            0, 1, 2);
+        $crate::impl_view_tuple!($viewseq <$($tp,)*>, $pod, $cx, $changeflags,
+            V0<$($tp,)*>, V1<$($tp,)*>, V2<$($tp,)*>, V3<$($tp,)*>;
+            0, 1, 2, 3);
+        $crate::impl_view_tuple!($viewseq <$($tp,)*>, $pod, $cx, $changeflags,
+            V0<$($tp,)*>, V1<$($tp,)*>, V2<$($tp,)*>, V3<$($tp,)*>, V4<$($tp,)*>;
+            0, 1, 2, 3, 4);
+        $crate::impl_view_tuple!($viewseq <$($tp,)*>, $pod, $cx, $changeflags,
+            V0<$($tp,)*>, V1<$($tp,)*>, V2<$($tp,)*>, V3<$($tp,)*>, V4<$($tp,)*>, V5<$($tp,)*>;
+            0, 1, 2, 3, 4, 5);
+        $crate::impl_view_tuple!($viewseq <$($tp,)*>, $pod, $cx, $changeflags,
+            V0<$($tp,)*>, V1<$($tp,)*>, V2<$($tp,)*>, V3<$($tp,)*>, V4<$($tp,)*>, V5<$($tp,)*>, V6<$($tp,)*>;
+            0, 1, 2, 3, 4, 5, 6);
+        $crate::impl_view_tuple!($viewseq <$($tp,)*>, $pod, $cx, $changeflags,
+            V0<$($tp,)*>, V1<$($tp,)*>, V2<$($tp,)*>, V3<$($tp,)*>, V4<$($tp,)*>, V5<$($tp,)*>, V6<$($tp,)*>, V7<$($tp,)*>;
+            0, 1, 2, 3, 4, 5, 6, 7);
+        $crate::impl_view_tuple!($viewseq <$($tp,)*>, $pod, $cx, $changeflags,
+            V0<$($tp,)*>, V1<$($tp,)*>, V2<$($tp,)*>, V3<$($tp,)*>, V4<$($tp,)*>, V5<$($tp,)*>, V6<$($tp,)*>, V7<$($tp,)*>, V8<$($tp,)*>;
+            0, 1, 2, 3, 4, 5, 6, 7, 8);
+        $crate::impl_view_tuple!($viewseq <$($tp,)*>, $pod, $cx, $changeflags,
+            V0<$($tp,)*>, V1<$($tp,)*>, V2<$($tp,)*>, V3<$($tp,)*>, V4<$($tp,)*>, V5<$($tp,)*>, V6<$($tp,)*>, V7<$($tp,)*>, V8<$($tp,)*>, V9<$($tp,)*>;
```

Comment:
I'm not sure if there's a less boilerplate heavy way to do this, but nesting repetitions is not possible unfortunately.